stats/opentelemetry/opentelemetry.go

```diff
@@ -220,6 +231,10 @@ type metricsInfo struct {
 
 	startTime time.Time
 	method    string
+
+	labelsReceived bool
```

Comment:
Should we replace this with a `nil` check on `labels`?

Reply:
The only change would be if GetLabels on a user set plugin option returns nil it would call it twice, but then there's no point of the plugin option. Function call in rare case worth code simplicitly.

Reply:
Hmmm...would this mean that it could generate labels from the trailers in the case of an RPC with both headers and trailers?  Would that "work correctly" - i.e. all metrics would get the labels that came out of the second call?

Reply:
The CSM Plugin Option specifically will never return nil labels if set. It'll always return at least the local labels, which will be "unknown" if nothing set in env.

Reply:
Right but it's using a "generic" (not exported) interface that anything else can implement.  So I'm fine with either:

1. Documenting that it must never return `nil` and (probably?) validating that it never returns nil.
2. Not documenting that but validating that it's always non-nil.
3. Actually allowing it, and letting the labels get determined by either one at the choice of the PluginOption.  It seems like it will actually "work correctly" in this case.

Of these I'm probably leaning toward (2).  I don't think we can make errors happen or anything, but we could replace `nil` with the empty map (`map[string]string{}`).

Reply:
Implemented 2.